From 8bf34d9a76d9c6eae60f96022932c2247fb7869b Mon Sep 17 00:00:00 2001
From: "Sachin George Thomas(Temp)" <sachth@codeaurora.org>
Date: Mon, 14 Oct 2019 17:02:29 +0530
Subject: [PATCH] use mixer_ctl_set_array api to set multiple arguments

Mixer control values , in case of multiple arguments , needs to be set
using mixer_ctl_set_array rather than individually setting using the
mixer_ctl_set_value api.

---
 tinymix.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/git/utils/tinymix.c b/git/utils/tinymix.c
index 53b5285..43cfefd 100644
--- a/git/utils/tinymix.c
+++ b/git/utils/tinymix.c
@@ -385,11 +385,13 @@ static void tinymix_set_value(struct mixer *mixer, const char *control,
                         num_values, num_ctl_values);
                 return;
             }
+            int tmp_arr[num_values];
             for (i = 0; i < num_values; i++) {
-                if (mixer_ctl_set_value(ctl, i, atoi(values[i]))) {
-                    fprintf(stderr, "Error: invalid value for index %u\n", i);
-                    return;
-                }
+                tmp_arr[i] = atoi(values[i]);
+            }
+            if (mixer_ctl_set_array(ctl, tmp_arr, num_values)) {
+               fprintf(stderr,"Invalid values");
+               return ;
             }
         }
     } else {
--

